backend/controllers/receptionMove/considerRisksReception.ts

```diff
@@ -0,0 +1,157 @@
+        offenderNumbersOfAllInReception
+      )
+
+      const otherOffenders = offendersInReception
+        .sort((left, right) => left.lastName.localeCompare(right.lastName, 'en', { ignorePunctuation: true }))
+        .map((offender) => ({
+          offenderNo: offender.offenderNo,
+          name: putLastNameFirst(offender.firstName, offender.lastName),
+          nonAssociation: offenderNumbersOfAllNonAssociations.includes(offender.offenderNo),
+          csraClassification:
+            offenderCsraStatus.find((statuses) => statuses.offenderNo === offender.offenderNo)?.classification ||
+            'Not entered',
+          displayCsraLink: offenderCsraStatus.find((statuses) => statuses.offenderNo === offender.offenderNo)
+            ?.assessmentComment,
+          alerts: offender.alerts
+            .map((alertCode) => alertFlagLabels.find((alertLabel) => alertLabel.alertCodes.includes(alertCode)))
+            .filter(Boolean)
+            .map((alertLabel) => ({ classes: alertLabel.classes, label: alertLabel.label }))
+            .sort((left, right) => left.label.localeCompare(right.label, 'en', { ignorePunctuation: true })),
+        }))
+
+      const nonAssociationsInEstablishment = await getNonAssociationsInEstablishment(
+        nonAssociations,
+        res.locals,
+        prisonApi
+      )
+      const sortedNonAssociationsInReceptionWithinCurrentEstablishment = nonAssociationsInEstablishment
+        .sort((left, right) => {
+          if (left.effectiveDate < right.effectiveDate) return 1
+          if (right.effectiveDate < left.effectiveDate) return -1
+          if (left.offenderNonAssociation.lastName > right.offenderNonAssociation.lastName) return 1
+          if (right.offenderNonAssociation.lastName > left.offenderNonAssociation.lastName) return -1
+          return 0
+        })
+        .filter((nonAssociationPrisoner) =>
+          offenderNumbersOfAllInReception.includes(nonAssociationPrisoner.offenderNonAssociation.offenderNo)
+        )
+
+      const { firstName, lastName } = await prisonApi.getDetails(res.locals, offenderNo)
+
+      const nonAssociationsRows = sortedNonAssociationsInReceptionWithinCurrentEstablishment?.map((nonAssociation) => ({
+        name: putLastNameFirst(
+          nonAssociation.offenderNonAssociation.firstName,
+          nonAssociation.offenderNonAssociation.lastName
+        ),
+        prisonNumber: nonAssociation.offenderNonAssociation.offenderNo,
+        type: nonAssociation.typeDescription,
+        selectedOffenderKey: `${formatName(firstName, lastName)} is`,
+        selectedOffenderRole: nonAssociation.reasonDescription,
+        otherOffenderKey: `${formatName(
+          nonAssociation.offenderNonAssociation.firstName,
+          nonAssociation.offenderNonAssociation.lastName
+        )} is`,
+        otherOffenderRole: nonAssociation.offenderNonAssociation.reasonDescription,
+        comment: nonAssociation.comments || 'Not entered',
+      }))
```

Comment:
If this is mostly copied from `backend/controllers/cellMove/viewNonAssociations.ts` then maybe it would be worth extracting it into a helper function?